### PR TITLE
Refactor helm storage setting

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: dockerfiles/hpcc/values.yaml [hunk 1]
```diff
@@ -2,8 +2,24 @@
...
+    #version: latest
+    root: "hpccsystems"    # change this if you want to pull your images from somewhere other than DockerHub hpccsystems
+    pullPolicy: IfNotPresent
   
-  dllserver:
+  # logging sets the default logging information for all components. Can be overridden locally
+  logging:
+    detail: 100
+
+  # Specify a defauleEsp to control which EclWatch service is returned from Std.File.GetEspURL, and other uses
```
Comment (ghalliday) [inline dockerfiles/hpcc/values.yaml:16]: unrelated: defauleEsp -> defaultEsp

File: dockerfiles/hpcc/values.yaml [hunk 2]
```diff
@@ -13,8 +29,9 @@ global:
     ##
     storageClass: ""
     # existingClaim: ""
```
Comment (ghalliday) [inline dockerfiles/hpcc/values.yaml:31]: Would benefit from documenting existingClaim and forcePermissions options.

**Comment by ghalliday** [comment]
That looks good to me.
@richardkchapman please review as well.

File: dockerfiles/hpcc/templates/_helpers.tpl [hunk 3]
```diff
@@ -156,9 +156,9 @@ imagePullPolicy: {{ .root.Values.global.image.pullPolicy | default "IfNotPresent
 {{- end -}}
 
 {{/*
-A kludge to ensure host mounted storage (e.g. for minikube or docker for desktop) has correct permissions for PV
```
Comment (richardkchapman) [inline dockerfiles/hpcc/templates/_helpers.tpl:159]: Not sure about removing the word host here - AFAIK this is only relevant when you are using single-host systems and mapping your mount point directly to your host system

Comment (ghalliday) [reply dockerfiles/hpcc/templates/_helpers.tpl:159]: or update the comment to cover the situation (?nfs) where it is required.

Comment (jakesmith) [reply dockerfiles/hpcc/templates/_helpers.tpl:159]: yes, I should have updated the comment.
The NFS case requires forcePermissions=true.
This is the draft PR for the NFS example : https://github.com/hpcc-systems/HPCC-Platform/pull/13773

File: dockerfiles/hpcc/values.yaml (hunk 1 above)
Comment (jakesmith) [reply dockerfiles/hpcc/values.yaml:16]: will fix

File: dockerfiles/hpcc/values.yaml (hunk 2 above)
Comment (jakesmith) [reply dockerfiles/hpcc/values.yaml:31]: have added some comments in new PR

**Comment by jakesmith** [comment]
replaced by https://github.com/hpcc-systems/HPCC-Platform/pull/13782

**Pull request closed without merging**